src/company_cmd.cpp

```diff
@@ -592,26 +592,29 @@ void StartupCompanies()
 	_next_competitor_start = 0;
 }
 
-/** Start a new competitor company if possible. */
-static bool MaybeStartNewCompany()
+/** Start new competitor companies if possible. */
+void MaybeStartNewCompany()
 {
-	if (_networking && Company::GetNumItems() >= _settings_client.network.max_companies) return false;
+	/* Allow multiple AIs to possibly start in the same tick. */
+	uint num_to_start = 0;
+	do {
+		if (_networking && Company::GetNumItems() + num_to_start >= _settings_client.network.max_companies) break;
 
-	Company *c;
+		Company *c;
 
-	/* count number of competitors */
-	uint n = 0;
-	FOR_ALL_COMPANIES(c) {
-		if (c->is_ai) n++;
-	}
+		/* count number of competitors */
+		uint n = num_to_start;
+		FOR_ALL_COMPANIES(c) {
+			if (c->is_ai) n++;
+		}
+		if (n >= (uint)_settings_game.difficulty.max_no_competitors) break;
+	} while (_settings_game.difficulty.max_no_competitors - ++num_to_start > 0 && AI::GetStartNextTime(num_to_start) == 0);
```

Comment:
This is really bad coding. Comparing, incrementing, and substracting in a single statement. Please don't punish future people like this ;)